packages/components/src/modifiers/hds-advanced-table-cell.ts

```diff
@@ -0,0 +1,79 @@
+import Modifier from 'ember-modifier';
+import type { ArgsFor, PositionalArgs } from 'ember-modifier';
+import { registerDestructor } from '@ember/destroyable';
+import { tracked } from '@glimmer/tracking';
+
+import {
+  didInsertGridCell,
+  updateTabbableChildren,
+} from './hds-advanced-table-cell/dom-management.ts';
+import { handleGridCellKeyPress } from './hds-advanced-table-cell/keyboard-navigation.ts';
+
+export interface HdsAdvancedTableCellModifierSignature {
+  Args: {
+    Named: {
+      handleEnableFocusTrap: () => void;
+      shouldTrapFocus: boolean;
+      setCellElement: (el: HTMLDivElement) => void;
+    };
+  };
+  Element: HTMLDivElement;
+}
+
+function cleanup(instance: HdsAdvancedTableCellModifier): void {
+  const { _observer } = instance;
+  if (_observer) {
+    _observer.disconnect();
+  }
+}
+
+export default class HdsAdvancedTableCellModifier extends Modifier<HdsAdvancedTableCellModifierSignature> {
+  // track shouldTrapFocus internally so the correct value is used when update tabbable children
+  @tracked private _shouldTrapFocus = false;
```

Comment:
question: do we need the `@tracked` decorator here? as far as I know, modifiers do not have a rendering lifecycle as components do
```suggestion
  private _shouldTrapFocus = false;
```

Reply:
You're correct! I updated it.